FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,19 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			if self == nil { return }
+			switch result {
+			case let .success((data, response)):
+				do {
+					let items = try FeedImagesMapper.map(data, response)
+					completion(.success(items))
+				} catch {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Following from my previous comment, by having the `map` function `throw` the correct error we don't have to overwrite it here. That way we could use a `do/catch` statement and complete with the `error` caught.

Or we could use `Result`'s [initializer](https://developer.apple.com/documentation/swift/result/3139399-init) that takes a throwing closure as parameter and returns `.success` with the value returned, or `.failure` with any thrown error:
```suggestion
				completion(Result { try FeedImageMapper.map(data, response) })
```

Reply:
Thanks for the review, I have an small doubt about how is this `Result` type working, is it using the following initializer
`public init(catching body: () throws -> Success)`?

I decided to make another change, instead of `throw`, I send the expected `FeedLoader.Result` I didn't try it at the first time because autocompletion was not working with that, should be that a bug?, I write it anyway and it worked so, I used that approach now.

Reply:
Ohhh gotcha! I was trying to make it work, but I missed the try keyword 👍

Reply:
Exactly @ fdorado985, the initializer takes a throwing closure as a parameter, and since it can `throw`, we need the `try` keyword to invoke it. And there's no problem returning a `Result`, that's a valid solution too. 

My comment was just to point in the direction that since Swift has ways of converting between `Result` and `throwing closures` (in the form of [`init(catching:)`](https://developer.apple.com/documentation/swift/result/3139399-init) and [`get()`](https://developer.apple.com/documentation/swift/result/3139397-get)), we should aim to use what's given by the language when possible, instead of rewriting the same logic.

Reply:
This is really cool, I'll update to use `Result.init(catching:)`, to keep this other way #TIL, thanks!